File: 1001-1500/1176_dietPlanPerformance.py
class Solution:
    def dietPlanPerformance(self, calories: List[int], k: int, lower: int, upper: int) -> int:
        calory = sum(calories[:k])
        point = 1 if calory > upper else -1 if calory < lower else 0
        for i in range(len(calories) - k):
            calory += calories[i + k] - calories[i]
            if calory > upper:
                point += 1
            elif calory < lower:
                point -= 1
        return point

File: 1001-1500/1176_dietPlanPerformance.java
class Solution {
    public int dietPlanPerformance(int[] calories, int k, int lower, int upper) {
        int point = 0;
        int windowSum = 0;
        for (int i=0; i<k; i++) windowSum += calories[i];
        point += (windowSum > upper) ? 1 : (windowSum < lower ? -1 : 0);
        for (int i=k; i<calories.length; i++) {
            windowSum += calories[i] - calories[i-k];
            point += (windowSum > upper) ? 1 : (windowSum < lower ? -1 : 0);
        }
        return point;
    }
}
